lib/galaxy/web/short_term_storage/__init__.py

```diff
@@ -261,7 +267,14 @@ def _directory(self, target: Union[UUID, ShortTermStorageTarget]) -> Path:
         return self._root.joinpath(*relative_directory)
 
     def _cleanup_if_needed(self, request_id: UUID):
-        request_metadata = self._load_metadata(self._directory(request_id), "request")
+        target_directory = self._directory(request_id)
+        if not target_directory.exists():
+            return  # Nothing to clean
+        request_metadata = self._load_metadata_safe(target_directory, "request")
+        if request_metadata is None:
+            # Delete if metadata is lost
+            self._delete(request_id)
```

Comment:
Don't you need to add `ignore_errors=True` on line 288 ?

Reply:
Nice catch, thank you Marius!